test/with_dummyserver/test_https.py

```diff
@@ -898,7 +901,11 @@ def test_tls_version_maximum_and_minimum(self) -> None:
                 conn = https_pool._get_conn()
                 try:
                     conn.connect()
-                    assert conn.sock.version() == self.tls_protocol_name  # type: ignore[attr-defined]
+                    if maximum_version != self.tls_version():
```

Comment:
Should we make this comparison more explicit with: `if maximum_version == TLSVersion.MAXIMUM_SUPPORTED`?